The addition of MADV_DONTDUMP to Solaris enables this code, containing
an incorrect cast that causes the newer compiler to complain. Fix the cast.
--- gcc-9.1.0.orig/libsanitizer/sanitizer_common/sanitizer_posix_libcdep.cc	2018-10-31 11:14:23.000000000 +0000
+++ gcc-9.1.0/libsanitizer/sanitizer_common/sanitizer_posix_libcdep.cc	2019-06-01 00:43:24.567973185 +0000
@@ -77,7 +77,7 @@
 
 bool DontDumpShadowMemory(uptr addr, uptr length) {
 #if defined(MADV_DONTDUMP)
-  return madvise((void *)addr, length, MADV_DONTDUMP) == 0;
+  return madvise((caddr_t)addr, length, MADV_DONTDUMP) == 0;
 #elif defined(MADV_NOCORE)
   return madvise((void *)addr, length, MADV_NOCORE) == 0;
 #else
